piker/ui/kivy/mouse_over.py

```diff
@@ -0,0 +1,84 @@
+"""Mouse over behaviour.
+
+Based on initial LGPL work by O. Poyen. here:
+https://gist.github.com/opqopq/15c707dc4cffc2b6455f
+"""
+
+
+from kivy.properties import BooleanProperty, ObjectProperty
+from kivy.core.window import Window
+
+
+class MouseOverBehavior(object):
+    """Mouse over behavior.
+
+    :Events:
+        `on_enter`
+            Fired when mouse enter the bbox of the widget.
+        `on_leave`
+            Fired when the mouse exit the widget.
+    """
+    hovered = BooleanProperty(False)
+    # Contains the last relevant point received by the Hoverable. This can
+    # be used in `on_enter` or `on_leave` in order to know where was dispatched
+    # the event.
+    border_point = ObjectProperty(None)
+    _widgets = []
+    _last_hovered = None
+
+    def __init__(self, **kwargs):
+        self.register_event_type('on_enter')
+        self.register_event_type('on_leave')
+        MouseOverBehavior._widgets.append(self)
+        super(MouseOverBehavior, self).__init__(**kwargs)
+        Window.bind(mouse_pos=self.on_mouse_pos)
```

Comment:
I assumed here this would only bind once, but if you have performance issues, i would check that assumption, as i'm not sure how bind plays with classmethod, maybe you have too many calls to `on_mouse_pos`. As mentionned, you could avoid it being called too often, by doing.
```
_on_mouse_pos = Clock.create_trigger(self.on_mouse_pos, timeout=0.01)
Window.bind(mouse_pos=_on_mouse_pos)
```

Reply:
Oh slick, I'll definitely try out the trigger.

Reply:
@tshirtman yeah so unfortunately this doesn't seem to work at all? The trigger doesn't call the `self.on_mouse_pos()` with the needed arguments - namely, the Window object and mouse coordinates.

Any suggestion to work around that? I looked at the [docs](https://kivy.org/doc/stable/api-kivy.clock.html#triggered-events) but don't really understand how triggers can pass through arguments to the underlying callback and it doesn't seem manually calling the event from a wrapper callback works at all?